powerauth-backend-tests/src/test/java/com/wultra/security/powerauth/test/v31/PowerAuthEnrollmentConfigurationTest.java

```diff
@@ -0,0 +1,115 @@
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.function.Predicate;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasProperty;
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Test for enrollment configuration endpoint.
+ *
+ * @author Lubos Racansky, lubos.racansky@wultra.com
+ */
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = PowerAuthTestConfiguration.class)
+@EnableConfigurationProperties
+class PowerAuthEnrollmentConfigurationTest {
+
+    @Autowired
+    private PowerAuthTestConfiguration config;
+
+    private EncryptStepModel encryptModel;
+
+    private final ObjectMapper objectMapper = new ObjectMapper().disable(SerializationFeature.FAIL_ON_EMPTY_BEANS);
+
+    @BeforeEach
+    void setUp() throws Exception {
+        encryptModel = new EncryptStepModel();
+        encryptModel.setApplicationKey(config.getApplicationKey());
+        encryptModel.setApplicationSecret(config.getApplicationSecret());
+        encryptModel.setMasterPublicKey(config.getMasterPublicKey());
+        encryptModel.setHeaders(new HashMap<>());
+        encryptModel.setResultStatusObject(config.getResultStatusObjectV31());
+        encryptModel.setVersion("3.1");
+    }
+
+    @Test
+    void testConfiguration() throws Exception {
+        encryptModel.setData(objectMapper.writeValueAsBytes("{}"));
+        encryptModel.setUriString(config.getEnrollmentServiceUrl() + "/api/configuration");
+        encryptModel.setScope("application");
+
+        final ObjectStepLogger stepLogger = new ObjectStepLogger(System.out);
+        new EncryptStep().execute(stepLogger, encryptModel.toMap());
+        assertTrue(stepLogger.getResult().isSuccess());
+        assertEquals(200, stepLogger.getResponse().getStatusCode());
+
+        final ConfigurationResponse response = stepLogger.getItems().stream()
+                .filter(isStepItemDecryptedResponse())
+                .map(StepItem::getObject)
+                .map(Object::toString)
+                .map(it -> safeReadValue(it, new TypeReference<ObjectResponse<ConfigurationResponse>>() { }))
+                .filter(Objects::nonNull)
+                .map(ObjectResponse::getResponseObject)
+                .findFirst()
+                .orElseThrow(() -> new AssertionError("error - no configuration found"));
+
+        assertThat(response, hasProperty("mobileApplication.android.minimalVersion", equalTo("1.4.0")));
+        assertThat(response, hasProperty("mobileApplication.android.currentVersion", equalTo("1.5.4")));
+        assertThat(response, hasProperty("mobileApplication.ios.minimalVersion", equalTo("1.5.4")));
+        assertThat(response, hasProperty("mobileApplication.iso.currentVersion", equalTo("2.0.0")));
```

Comment:
Typo `iso`. I wonder how this can work :)

Reply:
Well, it did not 😞 Too much refactor.